File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/Result.java
package io.github.sanyavertolet.edukate.backend.dtos;

import java.util.List;

public record Result(
        String id,
        String text,
        String notes,
        ResultType type,
        List<String> images
) {
    public Result {
        if (images == null) {
            images = List.of();
        }
    }

    public enum ResultType {
        FORMULA, TEXT, NUMERIC
    }

    public Result withImages(List<String> images) {
        return new Result(id, text, notes, type, images);
    }
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/Result.kt
package io.github.sanyavertolet.edukate.backend.dtos

data class Result(
    val id: String,
    val text: String,
    val notes: String,
    val type: ResultType,
    val images: List<String>,
) {
    fun withImages(images: List<String>): Result = copy(images = images)

    enum class ResultType {
        FORMULA,
        TEXT,
        NUMERIC,
    }
}
